src/AzSK.ADO/Framework/Abstracts/CommandBase.ps1

```diff
@@ -157,20 +157,50 @@ class CommandBase: AzSKRoot {
 		#the next two bug log classes have been called here as we need all the control results at one place for
 		#dumping them in json file and auto closing them(to minimize api calls and auto close them in batches)
 		#if bug logging is enabled and path is valid, create the JSON file for bugs
-		if($this.InvocationContext.BoundParameters["AutoBugLog"] -and [BugLogPathManager]::GetIsPathValid()){
+		#AutoBugLog Conditions
+		$isPartialScan=$false
+		$bugsClosed=$null
+		if($this.InvocationContext.BoundParameters["AutoBugLog"]){
 			if (([PartialScanManager]::ControlResultsWithBugSummary| Measure-Object).Count -gt 0)
 			{
 				$methodResult = [PartialScanManager]::ControlResultsWithBugSummary
+				# $bugsClosed=[PartialScanManager]::ControlResultsWithClosedBugSummary
+				$isPartialScan=$true
+			}
+			if (([PartialScanManager]::ControlResultsWithClosedBugSummary| Measure-Object).Count -gt 0)
+			{
+				$bugsClosed=[PartialScanManager]::ControlResultsWithClosedBugSummary
+				$isPartialScan=$true
+			}
+			 #If upc enabled dont call Auto close functions again
+			if(!$isPartialScan)
+			{
+				$AutoClose=[AutoCloseBugManager]::new($this.OrganizationContext.OrganizationName);
+				$AutoClose.AutoCloseBug($methodResult)
+				$bugsClosed=[AutoCloseBugManager]::ClosedBugs
+			}
+			if([BugLogPathManager]::GetIsPathValid()){
+					[PublishToJSONAndCSV]::new($methodResult,$folderPath,$bugsClosed)
 			}
-			[PublishToJSON]::new($methodResult,$folderPath)
 		}
-
-		#auto close passed bugs
-			if ($this.InvocationContext.BoundParameters["AutoBugLog"] -or $this.InvocationContext.BoundParameters["AutoCloseBugs"]) {
-			#call the AutoCloseBugManager
-			$AutoClose=[AutoCloseBugManager]::new($this.OrganizationContext.OrganizationName);
-			$AutoClose.AutoCloseBug($methodResult)
+		if($this.InvocationContext.BoundParameters["AutoCloseBugs"]){
+			if (([PartialScanManager]::ControlResultsWithClosedBugSummary| Measure-Object).Count -gt 0)
```

Comment:
What if command uses -autoclosebugs -autobuglog
Wouldn't this code be redundant?

Reply:
Will add condition to avoid redundancy

Reply:
Fixed